apps/aecore/src/aec_chain_state.erl

```diff
@@ -772,22 +772,16 @@ update_top_block_hash(Hash, State) ->
 apply_node_transactions([Node|Left], Trees, State) ->
     Txs = aec_blocks:txs(Node#node.content),
     Height = node_height(Node),
-    {ok, FilteredTxs, NewTrees} = aec_tx:apply_signed(Txs, Trees, Height),
-    assert_transactions_valid(FilteredTxs, Txs),
-    assert_state_hash_valid(NewTrees, Node),
-    apply_node_transactions(Left, NewTrees, State);
+    case aec_tx:apply_signed_strict(Txs, Trees, Height) of
```

Comment:
Can aec_blocks:new_with_state be refactored to use same function?

Reply:
No. That function has slightly different semantics.

In aec_blocks:new_with_state/3 we actually want to filter away the unwanted transactions for the candidate generation. Here, there should be no unwanted transactions.

A subtle but important difference.